Update issue templates and contribution guidelines to clarify who can work on what (#11419)
wagtail/wagtail: .github/ISSUE_TEMPLATE/BUG_REPORT.md

-   Django version: Look in your requirements.txt, or run `pip show django | grep Version`.
-   Wagtail version: Look at the bottom of the Settings menu in the Wagtail admin, or run `pip show wagtail | grep Version:`.
-   Browser version: You can use https://www.whatsmybrowser.org/ to find this out.

> ### Working on this

Reviewer: ```suggestion
### Contribute to this
```

Semantics I know, but I often try to avoid the term 'work' and keep to contributing.

Reviewer 2: @lb- I'd disagree here - the aim of this section is to hopefully catch the attention of people who would post a comment asking "can I work on this?". That's going to be more effective if we match the language people are already using.

(Also, to me "contributing" to an issue would include things like replicating the issue or suggesting technical approaches to it, which are welcome but not what this particular paragraph is about - we're not asking people to leave a comment before doing those things.)

Reviewer: Fair enough, happy to go with this direction then. Thanks for explaining.

<!--
  Do you have thoughts on skills needed?
  Are you keen to work on this yourself once the issue has been accepted?
  Please let us know here.
-->

> Anyone can work on this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), claim the issue once you’re ready to start.

Reviewer: ```suggestion
Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.
```
